src/NBitcoin/Enumeration.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace NBitcoin
+{
+    /// <summary>
+    /// Enumeration classes that enable all the rich features of an object-oriented language
+    /// <seealso cref="https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/microservice-ddd-cqrs-patterns/enumeration-classes-over-enum-types"/>
+    /// </summary
+    public abstract class Enumeration : IComparable
+    {
+        public string Name { get; private set; }
```

Comment:
Why would we need the name?

Reply:
We don't. It seems the solution was copypasted but not yet adapted for our needs